### PR TITLE
[FIX] Use the new product's UoM when checking for availability

### LINKED ISSUE #2559: Stock available checking still does the wrong thing with UoMs
In v7, @mart-e committed 7dbfcede361eb1e2c375b2a4693e6d3747f00ee6 some time ago to fix the error messages displayed when the available quantity is insufficient while changing the product on a quotation line, but it still lets the wrong UoM pass through in some cases.
To make it worst, it also leaves the initial UoM in the context, and that is a problem for custom addons which try to change the way the virtual stock is computed - those can not make the same magic as sale_stock does, they can only trust that the context makes sense.

I suggest we simply:
 - change the UoM in the context
 - and the use the new product's UoM instead of the UoM sent by the client

### PR DESCRIPTION
**Pull request by clonedagain**
Fixes #2559
This is fixes both the fact that user-facing warning messages are wrong,
and the fact that qty_available gets computed with the wrong UoM, which
is a problem for custom addons - i.e. https://github.com/numerigraphe/stock-logistics-warehouse/commit/ad4a88088da6c40b253f6c014c57629df556c8b4

### CODE REVIEW

**Comment by clonedagain** [comment]
@mart-e I've given it another try, I tried to apply your suggestion, can you tell me if it's better please?

File: addons/sale_stock/sale_stock.py [hunk 1]
```diff
@@ -600,6 +599,20 @@ def product_id_change(self, cr, uid, ids, pricelist, product, qty=0,
...
+        if 'value' in res and 'product_uom' in res['value']:
+            # The superclass changed the UoM, so we should base all further computations on the new UoM
+            uom_context = {'uom': res['value']['product_uom']}
+        # Otherwise, if the use's just selected a new UoM, use it
+        elif uom:
+            uom_context = {'uom': uom}
+        if uom_context is not None:
+           if context is None:
+                context = uom_context
+            else:
```
Comment (mart-e) [inline addons/sale_stock/sale_stock.py]: indentation error

Comment (clonedagain) [inline addons/sale_stock/sale_stock.py]: Fixed, sorry - not enough coffee I suppose.

**Comment by mart-e** [comment]
Thanks for the update, it looks good now. I have merged it at 5be3ac1fd08922a25f051263544b6642bb3b8ac2 (did a few changes to reduce the diff)

**Pull request closed without merging**